### DISCUSSION TITLE
terraform refresh

### DISCUSSION THREAD

**Discussion by advoretskiy15** [gruntwork-io/knowledge-base#672] [How do I]
Hello gruntwork. community!

I'm developing custom terraform provider for my own API. One field of the field in my terraform resource updates after 10-30 seconds. So the terraform state and actual state doesn't match and I can't write proper test.

I need to run `terraform refresh` before do validation.

Is there a way to do this in terratest?

---

<ins datetime="2023-02-28T01:38:21Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109936">Tracked in ticket #109936</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
I think this can be done by running function like `terraform.RunTerraformCommand(t, terraformOptions, "refresh")` 

https://github.com/gruntwork-io/terratest/blob/master/modules/terraform/cmd.go#L59
